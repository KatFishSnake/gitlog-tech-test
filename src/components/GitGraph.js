import React from "react";
import PropTypes from "prop-types";
import { Gitgraph, templateExtend, TemplateName } from "@gitgraph/react";
import styled from "styled-components";

import { graphColors } from "../config";

const GraphWrap = styled.div`
  // A way to remove annoying tooltips (again can be improved)
  pointer-events: none;

  // A way to remove strange non message labels (again can be improved)
  foreignObject {
    display: none !important;
  }
`;

const minimalisticTemplate = templateExtend(TemplateName.Metro, {
  colors: graphColors,
  branch: {
    lineWidth: 2,
    spacing: 12,
    label: { display: false },
  },
  commit: {
    message: { displayAuthor: false, displayHash: false, color: "transparent" },
    spacing: 24,
    dot: {
      size: 4,
      strokeWidth: 2,
    },
  },
  tag: {
    color: "transparent",
    strokeColor: "transparent",
    bgColor: "transparent",
  },
});

const GitGraph = ({ className, graphData }) => {
  return (
    <GraphWrap className={className}>
      <Gitgraph
        options={{
          template: minimalisticTemplate,
        }}
      >
        {(gitgraph) => {
          gitgraph.import(graphData);
        }}
      </Gitgraph>
    </GraphWrap>
  );
};

GitGraph.propTypes = {
  className: PropTypes.string,
  graphData: PropTypes.array,
};

export default GitGraph;
