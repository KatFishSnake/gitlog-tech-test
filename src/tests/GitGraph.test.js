import { render } from "@testing-library/react";

import mockedHistoryShort from "../mockedHistoryShort.json";
import GitGraph from "../components/GitGraph";

test("Renders GitGraph consistently", () => {
  const { container } = render(<GitGraph graphData={mockedHistoryShort} />);
  expect(container).toMatchSnapshot();
});
