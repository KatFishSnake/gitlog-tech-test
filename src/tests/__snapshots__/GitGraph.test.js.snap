// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders GitGraph consistently 1`] = `
<div>
  <div
    class="sc-bdfBwQ iGPnYJ"
  >
    <svg>
      <g
        transform="translate(10, 10)"
      >
        <g />
      </g>
    </svg>
  </div>
</div>
`;
